server/sonar-server/src/main/java/org/sonar/server/setting/ws/SettingsUpdater.java

```diff
@@ -42,49 +45,63 @@ public SettingsUpdater(DbClient dbClient, PropertyDefinitions definitions) {
     this.definitions = definitions;
   }
 
-  public void deleteGlobalSetting(DbSession dbSession, String propertyKey) {
-    delete(dbSession, propertyKey, Optional.empty());
+  public void deleteGlobalSettings(DbSession dbSession, String... settingKeys) {
+    deleteGlobalSettings(dbSession, asList(settingKeys));
   }
 
-  public void deleteComponentSetting(DbSession dbSession, String propertyKey, ComponentDto componentDto) {
-    delete(dbSession, propertyKey, Optional.of(componentDto));
+  public void deleteGlobalSettings(DbSession dbSession, List<String> settingKeys) {
+    checkArgument(!settingKeys.isEmpty(), "At least one setting key is required");
+    for (String propertyKey : settingKeys) {
```

Comment:
remark: you can use settingKeys.forEach, and call is settingKey or key but not propertyKey